src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1369,9 +1369,18 @@ private static bool IsCompilerAnalyzer(DiagnosticAnalyzer analyzer)
 
         public void Dispose()
         {
-            this.CompilationEventQueue?.TryComplete();
-            this.DiagnosticQueue?.TryComplete();
-            _queueRegistration.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
> The implementation of IDisposable.Dispose() is sealed. 

Implementation of an interface cannot be sealed, derived types can always re-implement any interface method. The class is internal, we control all concrete implementations of this class. Once a derived type adds a finalizer, we will deal with correct implementation of IDisposable, the finalizer will have to be written in a special way, nothing will just work because we added this code.